src/routes/tutorial/[slug]/state.js

```diff
@@ -1,5 +1,17 @@
 import { derived, writable } from 'svelte/store';
 
+/**
+ * The iframe sometimes takes focus control in ways we can't prevent
+ * while the editor is focused. Refocus the editor in these cases.
+ * This boolean tracks whether or not the editor should be refocused.
+ */
+export let preserve_editor_focus = false;
```

Comment:
This state is only relevant to the editor, I'd suggest keeping it in there. We already have `svelte:window` in `Editor.svelte`, which we can use to listen to messages as well.

Reply:
good catch, updated